_posts/2020-10-24-spring-boot-specification.md

```diff
@@ -0,0 +1,372 @@
+---
+title: "Intro to Specifications"
+categories: [spring-boot]
+date: 2020-10-24 00:00:00 +1100
+modified: 2020-10-24 20:00:00 +1100
+author: mmr
+excerpt: "All you need to know about spring boot specification"
+image:
+  auto: 0084-search
+---
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/spring-boot/specification" %}
+
+## What Is Specification?
+
+Specification is yet another tool at our disposal to perform queries in Spring Boot.
+
+Specification is an interface which is part of the Spring Data JPA 2.0 module, and it allows us to execute JPA 
+criteria API.
+
+```java
+interface Specification<T>{
+ 
+  Predicate toPredicate(Root<T> root, 
+                        CriteriaQuery<?> query, 
+                        CriteriaBuilder criteriaBuilder);
+
+}
+```
+
+When building [Criteria Query](https://docs.jboss.org/hibernate/stable/orm/userguide/html_single/Hibernate_User_Guide.html#criteria) we are required to build and manage `Root`, `CriteraQuery`, and `CriteriaBuilder` objects by ourselves. 
```

Comment:
Mention explicity that Specifications are built on top of the Criteria API. Also perhaps show a code example of a criteria query without Specifications and make a before/after comparison. This way the value of Specifications becomes more apparent.